component/http/middleware_test.go

```diff
@@ -368,3 +368,130 @@ func TestNewCompressionMiddleware_Headers(t *testing.T) {
 		})
 	}
 }
+
+func TestSelectEncoding(t *testing.T) {
+	testFn := func(hdr string, expected string, isErr bool) func(t *testing.T) {
+		return func(t *testing.T) {
+			result, err := parseAcceptEncoding(hdr)
+			assert.Equal(t, isErr, err != nil)
+			assert.Equal(t, expected, result)
+		}
+	}
+
+	t.Run("*", testFn("*", "*", false))
```

Comment:
Maybe it is better to use the std map based table driven tests here.
Check out the last test in this document: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
I don't actually think so. Table-tests are great but they have their downside. The best elaborate answer is here: https://medium.com/@cep21/closure-driven-tests-an-alternative-style-to-table-driven-tests-in-go-628a41497e5e
(psst: this post argues against exactly the post you gave)

Amongst other benefits of closure tests, for me personally, it is the ability to run a single test in IDE. It becomes less of an issue in a [newer Goland](https://www.jetbrains.com/go/promo/whatsnew/), but, hey, not everybody is using Goland.

Reply:
I see what you mean, but we have to be consistent here because not everybody follows the suggestion of the article and the community and patron is pretty much set up to use table driven.

Reply:
Ok, fair point about consistency. Maybe it's ok to have different styles in a project, but not inside a single test suite. I will change it to table tests